docs/guides/set_up_cluster.md

```diff
@@ -0,0 +1,184 @@
+
+### Startup and communication processes
+
+When started up in cluster mode, a node enters `ADMIN` mode, but is not automatically assigned a role like leader, follower, or learner. You must call `ADMIN SET NODE ROLE` to assign it a role. Once a leader node starts, it reads logs from the local disk to determine the metadata and data to read from shared storage.
+
+Once you set a node to follower or learner using `ADMIN SET NODE ROLE`, it registers with the leader node. Upon receiving the registration request, the leader node sends back its current log for the followers and learners to construct their data state from shared storage.
+
+### Keep-alive mechanism
+
+Once successfully registered with the leader node, a follower or learner starts sending periodic heartbeats to it. The leader node relies on these heartbeats to manage the connection status of each node. For example, if it does not receive heartbeats from a particular node for a specified time period, it sets that node's connection status to `timeout`.
+
+### Log synchronization
+
+When a transaction occurs, the leader node sends its log to both follower and learner nodes. The leader confirms the transaction's completion only after receiving confirmation that all its follower nodes have successfully persisted the log. While the leader also sends logs to learner nodes, it does not require confirmation from them.
+
+### Mode and role transition
+
+![uml](https://github.com/user-attachments/assets/276700ab-ee60-4a9c-a449-7e1d8ad5a92b)
+
+## Set up an Infinity cluster
+
+### Customize configuration files for cluster
+
+For *each* cluster node, you are required to prepare a customized configuration file to start it. Ensure that you properly set `server_mode`, `peer_ip`, `peer_port`, `storage_type`, and other related parameters. 
+
+1. Set `server_mode` to `"admin"`.
+
+2. Set `storage_type` to `"minio"`.
+
+3. Set `peer_ip` and `peer_port`.
+
+4. Update object storage-specific settings.
+
+5. Save your changes and start up Infinity using the customized configuration file.  
+
+   *When a cluster node starts, it automatically operates in `ADMIN` mode.*
+
+For further instructions on specifying a configuration file or setting parameters, see the [Configurations](https://infiniflow.org/docs/dev/configurations).
+
+### Set the leader node
+
+ A cluster can have only one leader node. If the cluster you start does not have a leader node, call `ADMIN SET NODE ROLE` to promote the node you just started, which currently operates in `ADMIN` mode, to leader. Below is an example code:
+
+````shell
+```shell
+curl --request POST \
+    --url http://localhost:23821/admin/node/current \
+    --header 'accept: application/json' \
+    --header 'content-type: application/json' \
+    --data ' {
+        "role" : "leader",
+        "name" : "Harry",
+    } '
+```
+````
+
```

Comment:
User can use following command to check current node role and status:

curl --request GET \
    --url http://localhost:23822/admin/node/current \
    --header 'accept: application/json'